packages/packages/top-bar/src/__tests__/index.test.tsx

```diff
@@ -0,0 +1,301 @@
+import { useState } from 'react';
+import '@testing-library/jest-dom';
+import {
+	injectIntoCanvasView,
+	injectIntoPrimaryAction,
+	registerAction,
+	registerLink,
+	registerToggleAction,
+} from '../locations/index';
+import { render, fireEvent, act } from '@testing-library/react';
+import ToolsMenuLocation from '../components/locations/tools-menu-location';
+import UtilitiesMenuLocation from '../components/locations/utilities-menu-location';
+import MainMenuLocation from '../components/locations/main-menu-location';
+import { MenuName } from '../locations/register-menu-item';
+import CanvasViewLocation from '../components/locations/canvas-view-location';
+import PrimaryActionLocation from '../components/locations/primary-action-location';
+
+describe( '@elementor/top-bar API', () => {
+	describe.each( [
+		{ component: MainMenuLocation, name: 'main' } as const,
+	] )( 'popover menu location: $name', ( { component: Component, name } ) => {
+		it( 'should render an action', () => {
+			// Arrange.
+			const onClick = jest.fn();
+
+			// Act
+			registerExampleAction( name, { onClick } );
+
+			// Assert.
+			const { getByText, getByRole } = render( <Component /> );
+
+			getByRole( 'button' ).click();
+
+			const button = getByText( 'Test' );
+
+			button.click();
+
+			expect( onClick ).toHaveBeenCalled();
+		} );
+
+		it( 'should render a toggle action', () => {
+			// Act.
+			registerExampleToggleAction( name );
+
+			// Assert.
+			const { getByRole, getByText } = render( <Component /> );
+
+			getByRole( 'button' ).click();
+
+			const menuItem = getByRole( 'menuitem' );
+			const menuItemButton = getByText( 'Test' );
+
+			expect( menuItem ).not.toHaveClass( 'Mui-selected' );
+
+			menuItemButton.click();
+
+			expect( menuItem ).toHaveClass( 'Mui-selected' );
+
+			menuItemButton.click();
+
+			expect( menuItem ).not.toHaveClass( 'Mui-selected' );
+			expect( menuItem ).toHaveAttribute( 'aria-disabled', 'true' );
+			expect( menuItemButton ).toHaveAttribute( 'disabled' );
```

Comment:
are those validations relevant also for the previous 2 states?

Reply:
It checks the the `disabled` attr works